### PR TITLE
More geth compatibility.

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: parity/main.rs [hunk 1]
```diff
@@ -285,45 +300,40 @@ impl Configuration {
...
-			}
-			public_address = Some(SocketAddr::from_str(a.as_ref()).unwrap_or_else(|_| die!("{}: Invalid listen/public address given with --public-address", a)));
-		}
 		(listen_address, public_address)
 	}
 
 	fn net_settings(&self, spec: &Spec) -> NetworkConfiguration {
 		let mut ret = NetworkConfiguration::new();
-		ret.nat_enabled = !self.args.flag_no_upnp;
+		ret.nat_enabled = self.args.flag_nat == "any" || self.args.flag_nat == "upnp";
```
Comment (gavofyork) [inline parity/main.rs]: ...and this? perhaps we should make an option from `flag_nat` and `flag_port` and pass directly into the network layer?

File: parity/main.rs [hunk 2]
```diff
@@ -285,45 +300,40 @@ impl Configuration {
...
+			r.extend(x.split(",").map(|s| Self::normalize_enode(s).unwrap_or_else(||die!("{}: Invalid node address format given for a boot node.", s))));
 		}
+		r
 	}
 
 	#[cfg_attr(all(nightly, feature="dev"), allow(useless_format))]
 	fn net_addresses(&self) -> (Option<SocketAddr>, Option<SocketAddr>) {
-		let mut listen_address = None;
-		let mut public_address = None;
+		let listen_address = Some(SocketAddr::new(IpAddr::from_str("127.0.0.1").unwrap(), self.args.flag_port));
```
Comment (gavofyork) [inline parity/main.rs]: @arkpar does this look ok?

Comment (arkpar) [inline parity/main.rs]: This will listen on loopback interface only. Use `"0.0.0.0"` to listen on all interfaces

File: parity/main.rs (hunk 1 above)
Comment (arkpar) [inline parity/main.rs]: If "any" means "auto" then setting `flag_nat` is ok. It will fallback to detecting interface address if upnp failed

**Pull request merged**